File: pathfinder-bukkit/src/main/java/de/cubbossa/pathfinder/listener/PlayerListener.java
package de.cubbossa.pathfinder.listener;

import de.cubbossa.pathfinder.misc.NamespacedKey;
import de.cubbossa.pathfinder.misc.PathPlayer;
import de.cubbossa.pathfinder.node.GraphEditorRegistry;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerQuitEvent;

public class PlayerListener implements Listener {

  @EventHandler
  public void onQuit(PlayerQuitEvent event) {
    PathPlayer<Player> player = PathPlayer.wrap(event.getPlayer());

    NamespacedKey currentlyEdited = GraphEditorRegistry.getInstance().getEdited(player);
    if (currentlyEdited != null) {
      GraphEditorRegistry.getInstance().<Player>getNodeGroupEditor(currentlyEdited)
          .thenAccept(e -> e.setEditMode(player, false));
    }
  }
}

File: pathfinder-bukkit/src/main/java/de/cubbossa/pathfinder/listener/PlayerListener.kt
package de.cubbossa.pathfinder.listener

import de.cubbossa.pathfinder.asPathPlayer
import de.cubbossa.pathfinder.node.GraphEditorRegistry
import org.bukkit.entity.Player
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerQuitEvent

class PlayerListener : Listener {

    @EventHandler
    fun onQuit(event: PlayerQuitEvent) {
        val player = event.player.asPathPlayer()

        val currentlyEdited = GraphEditorRegistry.getInstance().getEdited(player)
        currentlyEdited?.let {
            GraphEditorRegistry.getInstance()
                .getNodeGroupEditor<Player>(currentlyEdited)
                .thenAccept { it.setEditMode(player, false) }
        }
    }
}
